lib/utils/index.js

```diff
@@ -430,7 +446,21 @@ class Utils {
       return Utils.fileExistsSync(test);
     }
 
-    return Array.isArray(_source) && _source.length === 1 && Utils.fileExistsSync(_source[0]);
+    if (Array.isArray(_source) && _source.length === 1) {
+      // Check if the source is folder, and contains only a single file
```

Comment:
This check now has performance consequences because the (sync) read operation is now performed every time you're passing a folder as an argument, while the actual scenario which we are trying to avoid happens quite rarely. We need to find another way to deal with this.